test/plugins/candyMachineModule/uploadAssets.test.ts

```diff
@@ -0,0 +1,286 @@
+import test from 'tape';
+import spok from 'spok';
+
+import {
+  CandyMachineIsFullError,
+  Metaplex,
+  MetaplexFile,
+  mockStorage,
+  UploadedAsset,
+  useMetaplexFile,
+} from '../../../src';
+import {
+  amman,
+  killStuckProcess,
+  metaplex,
+  MOCK_STORAGE_ID,
+} from '../../helpers';
+import {
+  benchPng,
+  bridgePng,
+  createCandyMachineWithMaxSupply,
+  rockPng,
+  walrusPng,
+} from './helpers';
+import { Creator } from '@metaplex-foundation/mpl-token-metadata';
+import BN from 'bn.js';
+import { ammanMockStorage } from '@metaplex-foundation/amman-client';
+
+killStuckProcess();
+
+async function verifyProperlyUploaded(
+  t: test.Test,
+  mx: Metaplex,
+  uri: string,
+  asset: MetaplexFile,
+  creators: Creator[]
+) {
+  const { image, properties } = await mx.storage().downloadJson<any>(uri);
+  spok(
+    t,
+    properties.creators as Creator[],
+    creators.map((x) => ({
+      ...x,
+      address: x.address.toBase58(),
+      share: new BN(x.share).toNumber(),
+    }))
+  );
+
+  const metaplexFile = await mx.storage().download(image);
+  t.ok(
+    asset.buffer.equals(Buffer.from(metaplexFile.buffer)),
+    'asset.buffer === metaplexFile.buffer'
+  );
+}
+
+async function verifyUploadedAssets(
+  t: test.Test,
+  mx: Metaplex,
+  assets: MetaplexFile[],
+  uploadedAssets: UploadedAsset[],
+  creators: Creator[]
+) {
+  for (const x of uploadedAssets) {
+    const asset = assets.find((y) => y.displayName === x.name);
+    t.ok(asset != null, 'asset was named correctly');
+    await verifyProperlyUploaded(t, mx, x.uri, asset!, creators);
+  }
+}
+
+function setupMockStorage(mx: Metaplex) {
+  if (process.env.CI == null) {
+    mx.use(ammanMockStorage(MOCK_STORAGE_ID));
+  } else {
+    mx.use(mockStorage());
+  }
+}
```

Comment:
Should we set that up in the global `metaplex` helper function instead?

Reply:
I don't think so as this makes it clear that only for this test do we need mock storage.